Core/clim-basic/geometry/region-utilities.lisp

```diff
@@ -0,0 +1,1204 @@
+(in-package #:climi)
+
+
+;;; point and angle utilities
+
+;;; CLIM "native" coordinate system is left-handed (y grows down).
+;;; Angles are specified to grow in the counter-clockwise direction
+;;; (disregarding of the coordinate system), so we need to invert the
+;;; y coordinate in a call to atan.
+
+(defun atan* (x y)
+  ;; atan like we need it.
+  ;; For us, phi=0 along the x axis and angles are always between 0 and 2pi.
+  (let ((r (atan (float y pi) x)))
+    (if (< r 0) (+ r (* 2 pi)) r)))
+
+(declaim (inline angle-contains-angle-p))
+(defun angle-contains-angle-p (alpha omega delta)
+  (if (< alpha omega)
+      (coordinate-between* alpha delta omega)
+      (or (coordinate<= alpha delta)
+          (coordinate<= delta omega))))
+
+(declaim (inline angle-contains-point-p))
+(defun angle-contains-point-p (alpha omega x y)
+  (when (= 0 x y)
+    (return-from angle-contains-point-p t))
+  (angle-contains-angle-p alpha omega (atan* x (- y))))
+
+(defun colinear-p (p1 p2 p3)
+  (multiple-value-bind (x1 y1) (point-position p1)
+    (multiple-value-bind (x2 y2) (point-position p2)
+      (multiple-value-bind (x3 y3) (point-position p3)
+        (coordinate= (* (- x2 x1) (- y3 y2))
+                     (* (- x3 x2) (- y2 y1)))))))
+
+
+;;; line utilities
+(declaim (inline line-contains-point-p))
+(defun line-contains-point-p (x1 y1 x2 y2 px py)
+  (= (* (- py y1) (- x2 x1))
+               (* (- px x1) (- y2 y1))))
```

Comment:
Indentation. But also, why does this not use `coordinate=`?

Reply:
I don't know/remember, indentation shows that it was indeed coordinate= before. I take a guess that that must have been a typo. fixed